### DISCUSSION TITLE
How Strict Is the Requirement for PyTorch V1.5 or Later?

### DISCUSSION THREAD

**Discussion by brucejwittmann** [facebookresearch/esm#52] [Q&A] [answered]
I ask because I'm trying to reconcile a conda environment that also includes tensorflow v1.13, which only accepts CUDA Toolkit 10.0. Frustratingly, PyTorch versions over V1.4 do not accept CUDA Toolkit 10.0 (it needs to 9.2, 10.1, 10.2, or 11.0). I can get a working environment if I allow PyTorch V1.4, and my code runs to completion using esm models sourced from torch.hub. Your documentation lists PyTorch V1.5 or greater as a requirement, however, so I want to be sure I'm not potentially introducing any silent bugs by downgrading to 1.4.

Right now the models are just being used in eval mode, gradient off to extract sequence embeddings. I'm happy to provide code examples if it would help.

**Comment by joshim5** [comment, marked as answer]
Hi Bruce, we don't officially support PyTorch v1.4, but it is likely to still work. I don't see any methods added in the PyTorch v1.5 release notes that we include in this project. To be safe, I would recommend running the model in `.eval()` mode on two versions of PyTorch, running a few example tensors through the model, and confirming that the results are close.

**Reply by brucejwittmann** [reply]
Hey Josh, that sounds good. Thanks for your help!
